packages/core/src/api/guild.ts

```diff
@@ -0,0 +1,901 @@
+	 *
+	 * @param guildId - The id of the guild to fetch the auto moderation rules from
+	 */
+	public async getAutoModerationRules(guildId: Snowflake) {
+		return this.rest.get(Routes.guildAutoModerationRules(guildId)) as Promise<RESTGetAPIAutoModerationRulesResult>;
+	}
+
+	/**
+	 * Fetches an auto moderation rule for a guild
+	 *
+	 * @param guildId - The id of the guild to fetch the auto moderation rule from
+	 * @param ruleId - The id of the auto moderation rule to fetch
+	 */
+	public async getAutoModerationRule(guildId: Snowflake, ruleId: Snowflake) {
+		return this.rest.get(
+			Routes.guildAutoModerationRule(guildId, ruleId),
+		) as Promise<RESTGetAPIAutoModerationRuleResult>;
+	}
+
+	/**
+	 * Creates a new auto moderation rule for a guild
+	 *
+	 * @param guildId - The id of the guild to create the auto moderation rule from
+	 * @param data - The data for creating the auto moderation rule
+	 */
+	public async createAutoModerationRule(
+		guildId: Snowflake,
+		data: RESTPostAPIAutoModerationRuleJSONBody,
+		reason?: string,
+	) {
+		return this.rest.post(Routes.guildAutoModerationRules(guildId), {
+			reason,
+			body: data,
+		}) as Promise<RESTPostAPIAutoModerationRuleResult>;
+	}
+
+	/**
+	 * Edits an auto moderation rule for a guild
+	 *
+	 * @param guildId - The id of the guild to edit the auto moderation rule from
+	 * @param ruleId - The id of the auto moderation rule to edit
+	 * @param data - The data for editing the auto moderation rule
+	 * @param reason - The reason for editing the auto moderation rule
+	 */
+	public async editAutoModerationRule(
+		guildId: Snowflake,
+		ruleId: Snowflake,
+		data: RESTPatchAPIAutoModerationRuleJSONBody,
+		reason?: string,
+	) {
+		return this.rest.patch(Routes.guildAutoModerationRule(guildId, ruleId), {
+			reason,
+			body: data,
+		}) as Promise<RESTPatchAPIAutoModerationRuleJSONBody>;
+	}
+
+	/**
+	 * Deletes an auto moderation rule for a guild
+	 *
+	 * @param guildId - The id of the guild to delete the auto moderation rule from
+	 * @param ruleId - The id of the auto moderation rule to delete
+	 * @param reason - The reason for deleting the auto moderation rule
+	 */
+	public async deleteAutoModerationRule(guildId: Snowflake, ruleId: Snowflake, reason?: string) {
+		await this.rest.delete(Routes.guildAutoModerationRule(guildId, ruleId), { reason });
+	}
+
+	/**
+	 * Fetches a guild member
+	 *
+	 * @param guildId - The id of the guild
+	 * @param userId - The id of the user
+	 */
+	public async getMember(guildId: Snowflake, userId: Snowflake) {
+		return this.rest.get(Routes.guildMember(guildId, userId)) as Promise<RESTGetAPIGuildMemberResult>;
+	}
+
+	/**
+	 * Searches for guild members
+	 *
+	 * @param guildId - The id of the guild to search in
+	 * @param query - The query to search for
+	 * @param limit - The maximum number of members to return
+	 */
+	public async searchForMember(guildId: Snowflake, query: string, limit: number = 1) {
```

Comment:
```suggestion
	public async searchForMembers(guildId: Snowflake, query: string, limit: number = 1) {
```

Reply:
Also maybe just use the object type from types

Reply:
https://discord-api-types.dev/api/discord-api-types-v10/interface/RESTGetAPIGuildMembersSearchQuery